pallets/dex-pallet/src/exchange.rs

```diff
@@ -23,20 +23,52 @@ impl<T: Trait> Default for Exchange<T> {
 }
 
 impl<T: Trait> Exchange<T> {
+    // Avoid casting to float
+    fn sqrt(y: BalanceOf<T>) -> BalanceOf<T> {
```

Comment:
can we limit max value here?

Reply:
We shouldn`t have an overflow issue here.

Reply:
I implemented an overflow/underflow check here, though